src/v/config/configuration.cc

```diff
@@ -3717,6 +3717,56 @@ configuration::configuration()
       "Iceberg is enabled, do not change this value.",
       {.needs_restart = needs_restart::yes, .visibility = visibility::user},
       "redpanda-iceberg-catalog")
+  , iceberg_catalog_type(
+      *this,
+      "iceberg_catalog_type",
+      "Type of an Iceberg catalog that Redpanda will use to commit table "
+      "metadata updates. Supported types: 'rest', 'filesystem'",
+      {.needs_restart = needs_restart::yes, .visibility = visibility::user},
+      datalake_catalog_type::filesystem,
+      {datalake_catalog_type::rest, datalake_catalog_type::filesystem})
+  , iceberg_rest_catalog_endpoint(
+      *this,
+      "iceberg_rest_catalog_endpoint",
+      "URL of Iceberg REST catalog endpoint",
+      {
+        .needs_restart = needs_restart::yes,
+        .example = "http://hostname:8181",
+        .visibility = visibility::user,
+      },
```

Comment:
In the interest of having fewer configs.. wondering if an unset `iceberg_rest_catalog_endpoint` implies a filesystem catalog (thus removing the need for `iceberg_catalog_type`).

Reply:
IMO the straight number of configs isn't something to optimize. Number of tunables is a different story, but these categorical configs seem very straightforward